interface-manager/src/netns.rs

```diff
@@ -0,0 +1,119 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Open Network Fabric Authors
+
+//! Network namespace manipulation
+
+#![allow(missing_docs)]
+
+use nix::fcntl::OFlag;
+use nix::sched::CloneFlags;
+use nix::sys::stat::Mode;
+use std::future::Future;
+use std::os::fd::BorrowedFd;
+use std::path::Path;
+use tracing::error;
+
+/// Run an (async) function or closure in another network namespace.
+///
+/// This method will spawn a new thread and create a thread local tokio runtime to execute the
+/// provided method.
+///
+/// # Panics
+///
+/// * If we are unable to spawn a new thread
+/// * If we are unable to create a tokio runtime
+/// * If the provided function / closure panics
+/// * If the provided netns path is not legal Unicode
+pub fn in_netns<
+    Exec: (FnOnce() -> Fut) + Send + 'static,
+    Fut: Future<Output = Out> + Send,
+    Out: Send + 'static,
+>(
+    netns: &Path,
+    exec: Exec,
+) -> Out {
+    #[allow(clippy::expect_used)] // documented error case
+    let netns_str = netns
+        .to_str()
+        .expect("netns path not legal unicode")
+        .to_string();
+    let thread_name = format!("netns-{netns_str}");
+    #[allow(clippy::expect_used)]
+    std::thread::Builder::new()
+        .name(thread_name)
+        .spawn(move || {
+            #[allow(clippy::expect_used)] // the inability to swap to the other netns is fatal
+            #[allow(unsafe_code)] // uses external linux FFI
+            unsafe { swap_thread_to_netns(&netns_str) }.expect("failed to swap to netns");
+            #[allow(clippy::unwrap_used)] // the inability to start tokio is fatal
+            let tokio_runtime = tokio::runtime::Builder::new_current_thread()
+                .enable_io()
+                .enable_time()
+                .build()
+                .expect("failed to build tokio runtime");
```

Comment:
`unwrap()` no longer used, it's `expect()` now :slightly_smiling_face:

Reply:
good catch